### PR TITLE
New Admin User List Report

### PR DESCRIPTION
**Pull request by esoergel**
##### SUMMARY
Hoping to unblock the deletion of admin reports in https://github.com/dimagi/commcare-hq/pull/25105, I made a new user list report.

The search box is I think the main feature people will care about.  Right now it just uses the `search_string_query` logic, but I think that's a bit muddled - would be open to changing that.  My rollout plan is:

- [x] Roll this out close to as-is
- [ ] Confirm that the functionality meets minimum requirements to drop the other user-list report
- [ ] Drop the other user list report and all faceted admin reports (merge https://github.com/dimagi/commcare-hq/pull/25105)
- [ ] Make tweaks to the functionality as desired

##### FEATURE FLAG
Admin users only

##### PRODUCT DESCRIPTION
This introduces a new admin-focused user-list report.  It's pretty similar to the old one, but instead of the facets option on the lower left, this has a single search box.
image

### CODE REVIEW

**Comment by orangejenny** [comment]
yee haw!

🤠

File: corehq/apps/hqadmin/reports.py [hunk 1]
```diff
@@ -1331,3 +1335,80 @@ def rows(self):
...
+                self._format_date(user['date_joined']),
+                self._format_date(user['last_login']),
+                user['doc_type'],
+                user['is_superuser'],
+            ]
+
+    def _users_query(self):
+        query = (user_es.UserES()
+                 .remove_default_filters()
+                 .OR(user_es.web_users(), user_es.mobile_users()))
```
Comment (snopoke) [inline corehq/apps/hqadmin/reports.py]: why is this necessary? what other kind of users would this filter?

**Review by snopoke — APPROVED** [review]

File: corehq/apps/hqadmin/reports.py (hunk 1 above)
Comment (esoergel) [reply corehq/apps/hqadmin/reports.py]: there are `UnknownUser`, `AdminUser`, and `demo_user`.  Testing it out locally, I ran into issues because those "users" don't have the full set of fields available, so I figured it'd be easier to explicitly scope this to normal web and mobile users.

File: corehq/apps/hqadmin/reports.py [hunk 2]
```diff
@@ -1331,3 +1335,80 @@ def rows(self):
...
+        if 'search_string' in self.request.GET:
+            search_string = self.request.GET['search_string']
+            fields = ['username', 'first_name', 'last_name', 'phone_numbers',
+                      'domain_membership.domain', 'domain_memberships.domain']
```
Comment (esoergel) [inline corehq/apps/hqadmin/reports.py]: I'm considering switching this to search only `['base_username', 'first_name', 'last_name', 'phone_numbers']`, thus excluding matches based on domain.  If someone wants to constrain a search by domain, I imagine they'd use the domain-specific reports, not this one.  Any thoughts?
